### ISSUE TITLE
Can't `chattr` on pools that have `feature@project_quota=disabled`

### ISSUE THREAD

**Issue by heinrich5991** [zfsonlinux/zfs#7251]
### System information
Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | Arch Linux
Distribution Version    | rolling release
Linux Kernel                 | 4.15.3-2-ARCH
Architecture                 | x86_64
ZFS Version                  | 0.7.0-316_ge921f6508
SPL Version                  | 0.7.0-26_g48ef8ba

### Describe the problem you're observing

I can't change the immutable flag on files, whether it is set or not, on pools that have the `project_quota` feature disabled.

### Describe how to reproduce the problem

```
truncate -s 100M /tmp/test.img
sudo zpool create test /tmp/test.img -o feature@project_quota=disabled
sudo touch /test/foo
sudo chattr +i /test/foo
```

### Include any warning/errors/backtraces from the system logs
```
$ sudo chattr +i /test/foo
chattr: Operation not supported while setting flags on /test/foo
$ sudo strace chattr +i /test/foo
[...]
ioctl(3, FS_IOC_SETFLAGS, 0x7ffcc2bb962c) = -1 EOPNOTSUPP (Operation not supported)
[...]
```

**Comment by bunder2015** [comment]
can confirm that it works on ba779f7f, doesn't work on e921f650

**Comment by kpande** [comment]
it works for me on new pools that have the feature `enabled`.

**Comment by heinrich5991** [comment]
Yes, it works for me for pools that have the feature enabled as well.

**Comment by behlendorf** [comment]
@Nasf-Fan would you mind looking in to this issue.

**Comment by Nasf-Fan** [comment]
The issue will be fixed via #7265
